### PR TITLE
boot,image,overlord,partition: read/write boot variables in single operation

### PR DESCRIPTION
**Pull request by mvo5**

### CODE REVIEW

File: partition/grub.go [hunk 1]
```diff
@@ -63,28 +63,41 @@ func (g *grub) envFile() string {
...
 	}
 
-	return cfg.Get("", name)
+	return out, nil
 }
 
-func (g *grub) SetBootVar(name, value string) error {
+func (g *grub) SetBootVars(values map[string]string) error {
 	// note that strings are not quoted since because
 	// RunCommand() does not use a shell and thus adding quotes
```
Comment (niemeyer) [inline partition/grub.go]: `s/since//`
`s/RunCommand()/runCommand/` (no () as usual for Go docs)

File: partition/grub.go [hunk 2]
```diff
@@ -63,28 +63,41 @@ func (g *grub) envFile() string {
...
 	// stores them in the environment file (which is not desirable)
-	arg := fmt.Sprintf("%s=%s", name, value)
+	arg := ""
+	for k, v := range values {
+		arg += fmt.Sprintf("%s=%s ", k, v)
```
Comment (niemeyer) [inline partition/grub.go]: Given the comment above, this doesn't look right as it'll shove all the values in a single argument. The end result of `["set", "foo=1 bar=2"]` might well be `"foo"` being set to `"1 bar=2"`.

File: partition/grub_test.go [hunk 3]
```diff
@@ -88,9 +89,11 @@ func (s *PartitionTestSuite) TestSetBootVer(c *C) {
...
 
 	g := newGrub()
-	err := g.SetBootVar("key", "value")
+	err := g.SetBootVars(map[string]string{
+		"key": "value",
+	})
 	c.Assert(err, IsNil)
 	c.Assert(cmds, DeepEquals, [][]string{
-		{"/usr/bin/grub-editenv", g.(*grub).envFile(), "set", "key=value"},
+		{"/usr/bin/grub-editenv", g.(*grub).envFile(), "set", "key=value "},
```
Comment (niemeyer) [inline partition/grub_test.go]: Similarly, this also looks like a problem. We don't want that extra space at the end.

File: partition/grub.go (hunk 1 above)
Comment (mvo5) [inline partition/grub.go]: Thanks, fixed.

File: partition/grub.go (hunk 2 above)
Comment (mvo5) [inline partition/grub.go]: Indeed, silly me! this code was totally bogus.

File: partition/bootloader.go [hunk 4]
```diff
@@ -50,10 +49,10 @@ var (
 // bootloader
 type Bootloader interface {
 	// Return the value of the specified bootloader variable
-	GetBootVar(name string) (string, error)
+	GetBootVars(names []string) (map[string]string, error)
```
Comment (chipaca) [inline partition/bootloader.go]: Wouldn't making it `GetBootVars(names ...string)` make all the uses of `GetBootVars` nicer?

File: partition/grub.go [hunk 5]
```diff
@@ -63,28 +63,41 @@ func (g *grub) envFile() string {
 	return filepath.Join(g.Dir(), "grubenv")
 }
 
-func (g *grub) GetBootVar(name string) (string, error) {
+func (g *grub) GetBootVars(names []string) (map[string]string, error) {
+	out := map[string]string{}
```
Comment (chipaca) [inline partition/grub.go]: I'd suggest `out := make(map[string]string, len(names))` in this and the other non-mock `GetBootVars` (not that I'd mind if you did it in the mock ones as well)

**Comment by niemeyer** [comment]
Test failure is related to the changes:
```
FAIL: grub_test.go:84: TestSetBootVer.pN54_github_com_snapcore_snapd_partition.PartitionTestSuite
grub_test.go:104:
    c.Check(kwargs, DeepEquals, sort.StringSlice{"k1=v1", "k2=v2"})
... obtained sort.StringSlice = sort.StringSlice{"k2=v2", "k1=v1"}
... expected sort.StringSlice = sort.StringSlice{"k1=v1", "k2=v2"}
```

File: partition/bootloader.go (hunk 4 above)
Comment (mvo5) [inline partition/bootloader.go]: Indeed, let me fix this! Much nicer this way.

**Pull request merged**